candid_diff/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+// Logging:
+#[macro_use]
+extern crate log;
+extern crate env_logger;
+
+use std::rc::Rc;
+
+pub use ::candid::parser::value::IDLField as Field;
+pub use ::candid::parser::value::IDLValue as Value;
+
+pub use ::candid::types::{Field as TypeField, Label, Type};
+pub use ::candid::Nat;
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum VecEdit<R> {
+    /// insert into position, shifting the following (backward)
+    InsertValue(Nat, Value),
```

Comment:
What's a position, why it is a BigNat?

Reply:
Position within the vector to insert the given value.

Are you asking to use a different number type to describe the position, or questioning the meaning of this construct altogether?

Reply:
Yeah, then certainly don't need to use big number.